SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,256 @@
+//
+// Test for simd load constructor.
+// The test uses reference data and different alignment flags. Invokes simd
+// constructors in different contexts with provided reference data and alignment
+// flag.
+// It is expected for destination simd instance to store a bitwise same data as
+// the reference one.
+
+#include "common.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    simd<DataT, NumElems> source_simd(ref_data, alignment);
+    simd<DataT, NumElems> simd_by_init = simd<DataT, NumElems>(source_simd);
+    simd_by_init.copy_to(out, alignment);
+  }
+};
+
+// Descriptor class for the case of calling constructor in variable declaration
+// context.
+struct var_decl {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    simd<DataT, NumElems> simd_by_var_decl(ref_data, alignment);
+    simd_by_var_decl.copy_to(out, alignment);
+  }
+};
+
+// Descriptor class for the case of calling constructor in rvalue in an
+// expression context.
+struct rval_in_expr {
+  static std::string get_description() { return "rvalue in an expression"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    simd<DataT, NumElems> simd_by_rval;
+    simd_by_rval = simd<DataT, NumElems>(ref_data, alignment);
+    simd_by_rval.copy_to(out, alignment);
+  }
+};
+
+// Descriptor class for the case of calling constructor in const reference
+// context.
+class const_ref {
+public:
+  static std::string get_description() { return "const reference"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    call_simd_by_const_ref<DataT, NumElems>(
+        simd<DataT, NumElems>(ref_data, alignment), out, alignment);
+  }
+
+private:
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out, AlignmentT alignment) {
+    simd_by_const_ref.copy_to(out, alignment);
+  }
+};
+
+enum class alignment_type { element, vector, overaligned };
+
+template <typename DataT, int NumElems, alignment_type Alignment>
+std::size_t get_alignment_value() {
```

Comment:
when I has implemented the following code:
```
namespace alignment {

struct element {
  template <typename DataT, int NumElems> static size_t get_size() {
    return alignof(DataT);
  }
  template <typename DataT> static constexpr auto get_value() {
    return element_aligned;
  }
};

struct vector {
  template <typename DataT, int NumElems> static size_t get_size() {
    return alignof(simd<DataT, NumElems>);
  }
  template <typename DataT> static constexpr auto get_value() {
    return vector_aligned;
  }
};

struct overaligned {
  template <typename DataT, int NumElems> static size_t get_size() {
    return alignof(std::max_align_t);
  }
  template <typename DataT> static constexpr auto get_value() {
    return overaligned<alignof(std::max_align_t)>;
  }
};

} // namespace alignment
```
I obtain the following compilation error:
```
SYCL\ESIMD\api\functional\ctors\ctor_load_core.cpp:120:23: error: expected '(' for function-style cast or type construction
    return overaligned<alignof(std::max_align_t)>;
           ~~~~~~~~~~~^
SYCL\ESIMD\api\functional\ctors\ctor_load_core.cpp:120:50: error: expected expression
    return overaligned<alignof(std::max_align_t)>;
```

Reply:
Looks like name conflict with `using namespace sycl::ext::intel::experimental::esimd`

Reply:
I've changed namespace name to `alignment123` and error still produced

Reply:
As a suggestion - maybe we need to use "esimd" namespace alias for tests (the way you suggested before, but for all tests), to avoid any name conflicts in future? For example, if ESIMD API changes and adds some symbols

Reply:
I use it now in the headers and I am going to use an alias to `sycl::ext::intel::experimental::esimd` namespace

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/769/commits/9de67432a3606acda7db15caab8cc4c7ef03c3f4#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7R95-R124)

Reply:
So we are not going to use something like `esimd::simd<T,N>` and `esimd::overaligned<alignof(std::max_align_t)>` afterall? No complains, just a bit confused by comment and changes.

Reply:
yes